### PR TITLE
Run3-alca247B Add new options for getting correction factors - to be used to understand the impact on JetMET corrections

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Add new options for getting correction factors - to be used to understand the impact of variation in algorithms on JetMET corrections

#### PR validation:

Some of the options are tried on 2024C data

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Calibration/HcalCalibAlgos/macros/CalibCorr.C [hunk 1]
```diff
@@ -133,28 +133,53 @@ unsigned int truncateId(unsigned int detId, int truncateFlag, bool debug = false
...
+    if (subdet == 1) {
+      if (depth <= 2)
+        depth = 1;
+      else
+        depth = 2;
+    } else {
+      depth = 1;
+    }
+  } else if (truncate0 == 8) {
+    //Ignore depth index for depth > 2 in HB; depths 1, 2 considered as 1
```
Comment (perrotta) [inline Calibration/HcalCalibAlgos/macros/CalibCorr.C:165]:
```suggestion
    //Ignore depth index for depth > 2 in HE; depths 1, 2 considered as 1
```

File: Calibration/HcalCalibAlgos/macros/CalibPlotProperties.C [hunk 2]
```diff
@@ -71,12 +71,18 @@
 //   isRealData (bool)         = true/false for data/MC (default true)
 //   truncateFlag    (int)     = Flag to treat different depths differently (0)
 //                               both depths of ieta 15, 16 of HB as depth 1 (1)
-//                               all depths as depth 1 (2), all depths in HE
-//                               with values > 1 as depth 2 (3), all depths in
-//                               HB with values > 1 as depth 2 (4), all depths
-//                               in HB and HE with values > 1 as depth 2 (5),
-//                               for depth = 1 and 2, depth = 1, else depth
-//                                = 2 (6). (Default 0)
+//                               all depths as depth 1 (2), ignore all depth
```
Comment (perrotta) [inline Calibration/HcalCalibAlgos/macros/CalibPlotProperties.C:74]: Is there any reason why the description of the truncateFlag is different here with respect to the one in the other macros?

**Review by perrotta — COMMENTED** [review]
We must rerun the tests because of the issues in the baseline.
Before doing so, you could have a look at the following observations about your description fields

**Comment by perrotta** [comment]
please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by perrotta** [comment]
+alca

**Comment by rappoccio** [comment]
+1

**Pull request merged**